imap/mailbox.c

```diff
@@ -2648,8 +2648,9 @@ EXPORTED void mailbox_unlock_index(struct mailbox *mailbox, struct statusdata *s
     if (mailbox->local_cstate) {
         int r = conversations_commit(&mailbox->local_cstate);
         if (r) {
-            syslog(LOG_ERR, "IOERROR: Error committing to conversations database for mailbox %s: %s",
-                   mailbox_name(mailbox), error_message(r));
+            xsyslog(LOG_ERR, "IOERROR: Error committing to conversations database"
```

Comment:
Need a comma at the end of this line, i.e. between the fixed description and the format string for the varargs -- they should be separate args, not one concatenated string.  Right now the `%s`s are part of the description, and then it's treating `mailbox_name(mailbox)` as the format string and `error_message(r)` as the one argument to it.  And since `mailbox_name(mailbox)` is a string, and it doesn't contain any `%`s whose argument types might mismatch, the compiler fails to detect the misuse. :(

Reply:
This still needs fixing

```suggestion
            xsyslog(LOG_ERR, "IOERROR: Error committing to conversations database",
```